controllers/gce/controller/controller.go

```diff
@@ -275,13 +277,19 @@ func (lbc *LoadBalancerController) sync(key string) (err error) {
 	}
 	glog.V(3).Infof("Syncing %v", key)
 
-	ingresses, err := lbc.ingLister.List()
+	allIngresses, err := lbc.ingLister.ListAll()
 	if err != nil {
 		return err
 	}
-	nodePorts := lbc.tr.toNodePorts(&ingresses)
+	gceIngresses, err := lbc.ingLister.ListGCEIngresses()
+	if err != nil {
+		return err
+	}
+
+	allNodePorts := lbc.tr.toNodePorts(&allIngresses)
+	gceNodePorts := lbc.tr.toNodePorts(&gceIngresses)
 	lbNames := lbc.ingLister.Store.ListKeys()
-	lbs, err := lbc.ListRuntimeInfo()
+	lbs, err := lbc.ListGCERuntimeInfo()
```

Comment:
IMO, another nice change would be to plumb `gceIngresses` to `lbc.ListGCERuntimeInfo` and maybe rename it to `lbc.ToGCERuntimeInfo`. There would be no performance increase from this change, but the code would be clearer to follow and the function would appear less magical.  Up to you.

Reply:
Also named it toRuntimeInfo instead of toGCERuntimeInfo since it now has no single or multi cluster specific logic. Its a generic method that returns RuntimeInfo structs for the given ingress objests